src/dune_rules/inline_tests.ml

```diff
@@ -186,15 +187,92 @@ include Sub_system.Register_end_point (struct
         ~program:{ name; main_module_name = Module.name main_module; loc }
         ~linkages ~link_args ~promote:None
     in
-    let flags =
+    let partitions_flags : string list Action_builder.t option =
+      match
+        List.filter_map backends ~f:(fun backend ->
```

Comment:
Remind me again what are "backends" in this situations and why we can just append all their partition flags like this?

Reply:
Backends are inline test backends, We append all their partition flags similar to what we already do with their other flags (see `backend.Backend.info.flags`).

Reply:
I see. Do we have any tests with more than one backend? It might be worth adding such a test for partitions as well.

Reply:
I just discovered we have `test/blackbox-tests/test-cases/inline_tests/too-many-backends.t`.
I'm guessing that the list can have multiple entries when backend extend each other. (like ppx_inline_tests and ppx_expect).

Reply:
I've extended the test to have 2 backends, one extending the other, and only the second one using the partitioning mechanism.